projects/core/src/i18n/i18next/i18next-translation.service.ts

```diff
@@ -36,28 +36,46 @@ export class I18nextTranslationService implements TranslationService {
     // Otherwise, we the will trigger additional deferred change detection in a view that consumes the returned observable,
     // which together with `switchMap` operator may lead to an infinite loop.
 
-    const chunkName = this.translationChunk.getChunkNameForKey(key);
-    const namespacedKey = this.getNamespacedKey(key, chunkName);
+    const translationKeys = Array.isArray(key) ? key : [key];
+
+    const chunkNamesByKeys: Map<string, string> = new Map();
+    const namespacedKeys: string[] = [];
+
+    translationKeys.forEach((translationKey) => {
+      const chunkName =
+        this.translationChunk.getChunkNameForKey(translationKey);
+      const namespacedKey = this.getNamespacedKey(translationKey, chunkName);
+      namespacedKeys.push(namespacedKey);
+      chunkNamesByKeys.set(translationKey, chunkName);
+    });
 
     return new Observable<string>((subscriber) => {
       const translate = () => {
         if (!this.i18next.isInitialized) {
           return;
         }
-        if (this.i18next.exists(namespacedKey, options)) {
-          subscriber.next(this.i18next.t(namespacedKey, options));
-        } else {
-          if (whitespaceUntilLoaded) {
-            subscriber.next(this.NON_BREAKING_SPACE);
-          }
-          this.i18next.loadNamespaces(chunkName, () => {
-            if (!this.i18next.exists(namespacedKey, options)) {
-              this.reportMissingKey(key, chunkName);
-              subscriber.next(this.getFallbackValue(namespacedKey));
+
+        // make sure chunks are loaded
+        let namespacesLoaded = false;
+
+        // loadNamespaces has cache under the hood, so it won't load the same chunk twice
+        // when namespaces are already loaded, the callback will be called synchronously:
+        this.i18next.loadNamespaces(
+          Array.from(chunkNamesByKeys.values()),
+          () => {
+            namespacesLoaded = true;
+            if (this.i18next.exists(namespacedKeys, options)) {
+              subscriber.next(this.i18next.t(namespacedKeys, options));
             } else {
-              subscriber.next(this.i18next.t(namespacedKey, options));
+              this.reportMissingKey(chunkNamesByKeys);
+              subscriber.next(this.getFallbackValue(namespacedKeys));
             }
-          });
+          }
+        );
+
+        // if namespaces are not loaded yet, we can emit a non-breaking space
+        if (!namespacesLoaded && whitespaceUntilLoaded) {
+          subscriber.next(this.NON_BREAKING_SPACE);
         }
```

Comment:
Summary of the refactor, for future reference:

Previously
- let's check if a key can be resolved (aka "if exists"). if yes then let's emit it
- if not, then let's load a namespace chunk for this key. and in the meantime let's emit temporarily the NON_BREAKING_SPACE
- when namespace is eventually loaded (can be sync or async) and if the key still can't be resolved let's emit a fallback value

Now:
- first, let's load the namespace (can be sync or async)
- when the namespace is eventually loaded, check if the key can be resolved (aka "if exists"). if yes, let's emit it
- if not, let's emit the fallback value
Note: If retrieving the namespace wasn't instant (meaning: it was asynchronous), then in the meantime let's emit temporarily the NON_BREAKING_SPACE.

This refactor should not change the behavior seen by an external observer of this Observable.
It allowed us to more eaisly **introduce fallback keys** and **preload namespaces for all the fallback keys upfront**